src/opt/ReplaceHalfMaskedMemOps.h

```diff
@@ -0,0 +1,29 @@
+/*
+  Copyright (c) 2024, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#pragma once
+
+#include "ISPCPass.h"
+
+namespace ispc {
+
+// This pass transforms some masked loads and stores into their half-width
+// vector unmasked counterparts. This is done to improve the performance of
+// the generated code in cases when only half of the registers are used.
+// The masked.load and masked.store intrinsics are directly mapped to machine
+// instructions with the specified full width of vector values being loaded or
+// stored. This transformation allows the backend to generate shorter vector
+// loads and stores avoidind extra spills.
```

Comment:
Is it only about shorter memory ops or it's also about short math operations? I assumed that it's both.

Also, what do you mean by "spill" here? It's reads/writes of the user visible memory, while spills refer to storing/restoring from temporary memory location when register allocator runs out of registers.

Reply:
Changed to "This transformation allows the backend to generate shorter vector memory operations and corresponding math operations avoiding extra spills of temporal values to memory"